net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/config/meta/CheckUtil.java

```diff
@@ -0,0 +1,84 @@
+package net.sf.eclipsecs.core.config.meta;
+
+import java.util.Set;
+
+import com.puppycrawl.tools.checkstyle.PackageNamesLoader;
+import com.puppycrawl.tools.checkstyle.PackageObjectFactory;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import net.sf.eclipsecs.core.CheckstylePlugin;
+
+public final class CheckUtil {
+    private CheckUtil() {
+    }
+
+    public static Object getCheck(String checkName) {
+        final ClassLoader classLoader = CheckstylePlugin.getDefault()
+                .getAddonExtensionClassLoader();
+        try {
+            final Set<String> packageNames = PackageNamesLoader.getPackageNames(classLoader);
+            return new PackageObjectFactory(packageNames, classLoader)
+                    .createModule(checkName);
+        }
+        catch (CheckstyleException ex) {
+            throw new IllegalStateException("exception occured during load of " + checkName, ex);
+        }
+    }
+
+    public static String getAcceptableTokens(String checkName) {
+        final Object checkResult = getCheck(checkName);
+        String result = null;
+        if (AbstractJavadocCheck.class.isAssignableFrom(checkResult.getClass())) {
+            final AbstractJavadocCheck javadocCheck = (AbstractJavadocCheck) checkResult;
+            result = getTokenText(true, javadocCheck.getAcceptableJavadocTokens(),
+                    javadocCheck.getRequiredJavadocTokens());
+        }
+        else if (AbstractCheck.class.isAssignableFrom(checkResult.getClass())) {
```

Comment:
please throw exception in  `else` block to make it clear that system in damaged state and this method is executed in wrong context on Check (not a some other module) that is not following requirements for hierarchy of classes.

Reply:
Done